### ISSUE TITLE
"Copy to Clipboard" not working in FF57 MTH 2.0.4 

### ISSUE THREAD

**Issue by eberhardweber** [piroor/multipletab#178]
## Short description

The "Copy to Clipboard" functionality is not currently working in FF57 MTH 2.0.4. None of the copying features add desired content to clipboard.

## Steps to reproduce

1. Select tab(s) in MTH
2. Select "Copy to Clipboard"
3. Select any of the 5 available types of copy
4. Nothing gets added to clipboard

## Expected result

Selected type of tab URL copy should happen by MTH

## Actual result

Nothing is copied to clipboard by MTH

## Environment

 * Platform (OS): Windows 7 64-bit SP1
 * Version of Firefox: Firefox 57 Stable 64-bit
 * Version (or revision) of Multiple Tab Handler: 2.0.4

**Comment by piroor** [comment]
Works for me, but there is a known restriction: you'll fail to run this command when all tabs are special tabs (about:addons, about:config, https://addons.mozilla.org/, etc.) Currently we need to run a code to copy text inside a tab, and addons cannot run such codes inside those special pages. See also:

https://bugzilla.mozilla.org/show_bug.cgi?id=1272869
https://bugzilla.mozilla.org/show_bug.cgi?id=1344410

**Comment by eberhardweber** [comment]
I clearly rushed to make a bug report without testing as I had completely forgotten addons.mozilla.org is also on the restricted list. 

I've been recommending add-ons non-stop to new FF57 users and had set "privacy.resistFingerprinting.block_mozAddonManager" to true in about:config on my previous profile.

Now that I did it again on my new profile, MTH works on addons.mozilla.org again so this isn't anything to do with your addon!

**Comment by joshgold22** [comment]
Is there any way for MTH to at least *detect* when it's not going to work, or when it has failed to work.  A popup would be much better than going back and forth.  I often re-select (sometimes carefully selecting many tabs) and try again, because I wonder if I misclicked.  Also because it's unpredictable, I (at least until having read this bug report) try 4 or 5 times, perhaps selecting different subsets.

I've seen it not work when I've only had a couple selected that are not "special tabs" (but with a few special tabs elsewhere in the tree.)  I've also seen it work fine with 150 tabs selected, including some special tabs in and not in the selection.

As an additional experiment, I closed some LeechBlock tabs.  This made the extension Copy All Tabs able to work, but not MTH.  Then I closed the rest of the special tabs (blank, about:addons, and about:config) and MTH still would not work, no matter which subset of the 8 tabs open that I tried.  (MTH 2.0.10)
